File: 1_b.swift
let target: Int = 2020

var nums = [Int]()
while let tmp = readLine(), let i = Int(tmp) {
    nums.append(i)
}

nums.sort()

for (i, num) in nums.enumerated() {
    var l = i+1
    var r = nums.count - 1
    while l < r {
        let cand = [num, nums[l], nums[r]]
        let sum = cand.reduce(0, +)
        if sum == target {
            print("found", cand.reduce(1, *))
            break
        } else if sum < target {
            l += 1
        } else {
            r -= 1
        }
    }
}

File: 1_b.py
nums = []
target = 2020
while True:
    try:
        nums.append(int(input()))
    except:
        break
        
nums.sort()

for i, cur in enumerate(nums):
    l = i+1
    r = len(nums)-1
    
    while(l < r):
        sum3 = sum([cur, nums[l], nums[r]])
        if (sum3 == target):
            print("found", cur * nums[l] * nums[r])
            break
        elif (sum3 < target):
            l += 1
        else:
            r -= 1
        
